commit 678e08735bf5c2dc9b6e120e7d08fc4249654fd7 (refs/changes/52/52/2)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2016-07-11T12:49:38-07:00 (3 years, 3 months ago)
    
    ZAPI-745: Revert from restify 4.x to restify 2.7
    
    This reverts commit 3f90d046b790f03236628e02ae02b6111bc0f0eb.
    
    Reviewed by: Trent Mick <trentm@gmail.com>

diff --git a/lib/vmapi.js b/lib/vmapi.js
index d6ab3e3..e3da883 100644
--- a/lib/vmapi.js
+++ b/lib/vmapi.js
@@ -413,9 +413,7 @@ VMAPI.prototype.listen = function (callback) {
 /*
  * Force JSON for all accept headers
  */
-function formatJSON(req, res, body, callback) {
-    var formattedJson;
-
+function formatJSON(req, res, body) {
     if (body instanceof Error) {
         // snoop for RestError or HttpError, but don't rely on
         // instanceof
@@ -442,12 +440,10 @@ function formatJSON(req, res, body, callback) {
         // used to properly set the Content-Length and Content-MD5 headers, but
         // no data should actually be sent as part of the response's body. This
         // is all according to RFC 2616.
-        formattedJson = '';
+        return ('');
     } else {
-        formattedJson = data;
+        return (data);
     }
-
-    callback(null, formattedJson);
 }
 
 
diff --git a/package.json b/package.json
index 4f4e8ae..df1b8e3 100644
--- a/package.json
+++ b/package.json
@@ -18,7 +18,7 @@
     "moray": "git+ssh://git@github.com:joyent/node-moray.git#b84ef0e",
     "nodeunit": "0.9.1",
     "once": "^1.3.3",
-    "restify": "4.1.1",
+    "restify": "2.7.0",
     "sdc-clients": "git+ssh://git@github.com:joyent/node-sdc-clients.git#c962959",
     "sigyan": "0.2.0",
     "sprintf": "0.1.1",
